### PR TITLE
Fixed RuntimeType.GetConstructorCandidates("") to return 0 elements.

### PR DESCRIPTION
**Pull request by steveisok**
https://github.com/mono/mono/issues/16010

The current behavior for GetConstructorCandidates when the name is null, "",
or "*" is to return the set of constructors for the type. For "", .NET Core and
.NET Framework return 0 elements.  This PR adds a check for "" to match that
behavior.

### CODE REVIEW

File: mcs/class/referencesource/mscorlib/system/rttype.cs [hunk 1]
```diff
@@ -2858,17 +2858,24 @@ internal RuntimeConstructorInfo GetSerializationCtor()
...
         {
             bool prefixLookup, ignoreCase;
             MemberListType listType;
+
             RuntimeType.FilterHelper(bindingAttr, ref name, allowPrefixLookup, out prefixLookup, out ignoreCase, out listType);
 
-#if MONO
-            if (!string.IsNullOrEmpty (name) && name != ConstructorInfo.ConstructorName && name != ConstructorInfo.TypeConstructorName)
+#if MONO            
+            if ((!prefixLookup && name != null && name.Length == 0) ||
```
Comment (marek-safar) [inline mcs/class/referencesource/mscorlib/system/rttype.cs]: could be just `(!prefixLookup && name?.Length == 0`) ||  for better readability

**Comment by steveisok** [comment]
@monojenkins build Linux ARMv5

**Comment by steveisok** [comment]
@monojenkins build failed

**Comment by marek-safar** [comment]
@monojenkins backport 2019-06

**Pull request merged**